docs/research/api-testing-plan

```diff
@@ -0,0 +1,99 @@
+# API Testing Research
+
+## Measuring Node Runtime Performance
+
+1. Using performance Timers within codebase
+
+- Date.now()
+- console.time()
+- performance.now() <- most precise option!!
+- Unit Tests
+- @vnajack has set up timers into the codebase, we might need to update it a bit more
```

Comment:
The timers are very basic and can be switched out for sure! I just wanted to make sure we started implementing them!

Reply:
sweet. It's definetly better than nothing and adds some accurate measuring data!